compiler/jsgen.nim

```diff
@@ -1434,8 +1434,19 @@ proc genAddr(p: PProc, n: PNode, r: var TCompRes) =
     of nkConv:
       genAddr(p, n[0], r)
     of nkStmtListExpr:
-      if n.len == 1: gen(p, n[0], r)
-      else: internalError(p.config, n[0].info, "genAddr for complex nkStmtListExpr")
+      let x = n[0]
+      if n.len == 1:
+        let isExpr = not isEmptyType(x.typ)
+        if x.kind == nkStmtListExpr and isExpr and n.kind in {nkHiddenAddr, nkAddr}:
```

Comment:
You already know that `x.kind == nkStmtListExpr` here.

Reply:
You also know that `n.kind in {nkHiddenAddr, nkAddr}` because it's called `genAddr`.